prov/tcp/src/tcpx_ep.c

```diff
@@ -628,6 +670,45 @@ static struct fi_ops tcpx_pep_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static int tcpx_pep_setname(fid_t fid, void *addr, size_t addrlen)
+{
+	struct tcpx_pep *tcpx_pep;
+
+	if ((addrlen != sizeof(struct sockaddr_in)) &&
+	    (addrlen != sizeof(struct sockaddr_in6)))
+		return -FI_EINVAL;
+
+	tcpx_pep = container_of(fid, struct tcpx_pep,
+				util_pep.pep_fid);
+
+	if (tcpx_pep->sock != INVALID_SOCKET) {
+		ofi_close_socket(tcpx_pep->sock);
+		tcpx_pep->sock = INVALID_SOCKET;
+	}
+
+	if (tcpx_pep->info.src_addr)
+		free(tcpx_pep->info.src_addr);
```

Comment:
Please add setting src_addrlen = 0 when freeing src_addr, so that they remain in sync.